doc/api/n-api.md

```diff
@@ -1612,13 +1650,15 @@ added: REPLACEME
 
 ```c
 NAPI_EXTERN napi_status napi_remove_async_cleanup_hook(
-    napi_env env,
```

Comment:
I removed the need to pass `napi_env` here because we wish to discourage JS execution in environment cleanup hooks and this way it should prevent add-ons from having to store the `env` for the sole purpose of calling this function from a `uv_close` callback.

We store `napi_env` internally and it's guaranteed to be usable for the lifetime of a `remove_handle`, because we `Ref()` it during its construction and `Unref()` it asynchronously during its destruction.